tests/test_tracing.py

```diff
@@ -83,6 +84,22 @@ def test_instrument_decorator_with_name_and_attributes():
     assert current_span.attributes == {"foo": "bar"}  # type: ignore
 
 
+def test_instrument_decorator_parent_attributes(settings):
+    @instrument(span_name="child", attributes={"foo": "bar"})
+    def child(): ...
```

Comment:
oh nice, yeah, I see how that is a bit awkward. 

But I've used this pattern (decorating a closure) before when testing decorators, I think it works well.